File: frontend/src/types/analysis.ts
export type BoundingBox = {
  x1: number;
  y1: number;
  x2: number;
  y2: number;
};

export type Detection = {
  label: string;
  confidence: number;
  bbox: BoundingBox;
};

export type Analysis = {
  id: number;
  image_name: string;
  detections: Detection[];
  risk_level: "LOW" | "MEDIUM" | "HIGH" | string;
  risk_score: number;
  reasons: string[];
  created_at: string;
};

File: backend/app/schemas/analysis.py
from datetime import datetime

from pydantic import BaseModel


class BoundingBox(BaseModel):
    x1: float
    y1: float
    x2: float
    y2: float


class DetectionResponse(BaseModel):
    label: str
    confidence: float
    bbox: BoundingBox


class AnalysisResponse(BaseModel):
    id: int
    image_name: str
    detections: list[DetectionResponse]
    risk_level: str
    risk_score: float
    reasons: list[str]
    created_at: datetime
